src/xcode/ENA/ENA/Source/Models/Metadata/Client Metadata/__tests__/ClientMetadataTests.swift

```diff
@@ -0,0 +1,43 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+@testable import ENA
+import XCTest
+
+class ClientMetadataTests: XCTestCase {
+
+	func testGIVEN_ClientMetadat_WHEN_AnalyticsCollectIsCalled_THEN_ClientMetadataAreLoggedCorrect() {
+		// GIVEN
+		let mockStore = MockTestStore()
+		Analytics.setupMock(store: mockStore)
+		mockStore.isPrivacyPreservingAnalyticsConsentGiven = true
+
+		XCTAssertNil(mockStore.clientMetadata, "Client metadata should be initially nil")
+		let appVersionParts = Bundle.main.appVersion.split(separator: ".")
+		guard appVersionParts.count == 3,
+			  let majorAppVersion = Int(appVersionParts[0]),
+			  let minorAppVersion = Int(appVersionParts[1]),
+			  let patchAppVersion = Int((appVersionParts[2])) else {
+			return
+		}
+		let expectedAppVersion = Version(major: majorAppVersion, minor: minorAppVersion, patch: patchAppVersion)
+		// iOSVersion
+		let expectediosVersion = Version(
+			major: ProcessInfo().operatingSystemVersion.majorVersion,
+			minor: ProcessInfo().operatingSystemVersion.minorVersion,
+			patch: ProcessInfo().operatingSystemVersion.patchVersion
+		)
```

Comment:
I think I've seen similar code somewhere else 🤷‍♂️ If that's the case we should create s simple struct to wrap `ProcessInfo().operatingSystemVersion` and all comparisons.

Reply:
Because this is a high bug ticket, I'd liked not to refactor that because I just moved that code around. But I can create a chore ticket that this will not be forgotten.

Reply:
Fine by me, I did not expect this to happen right now. Sorry for being ambiguous.